### PR TITLE
execute a plan against the terraform destroy & apply command

### PR DESCRIPTION
**Pull request by maganti07**
tf:plan 
1.-DplanOutputFile=create.json should create a plan file.
 2. Added planOutputFile parameter in maven-plugin
tf:apply
 1. Added plan action support
 2. If both plan and var files specified, only plan will be used
tf:destroy
 1. if Plan specifies, apply will be used to destroy using plan files
 2. added plan support and destroys with plan file
updated readme and test cases

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -204,7 +205,9 @@ Optional Parameters:
 | target      | Number | A resource address to target                                                                                       |
 | noColor     | Any    | If this property exists, the -no-color flag is set                                                                 |
 | tfRootDir   | String | A terraform config directory to destroy; defaults to current directory                                             |
+| plan        | String | A terraform plan to destroy; if both plan and tfRootDir are specified, only plan is used                           |
```
Comment (Clayton7510) [inline README.md:208]: Shouldn't plan only be for the 'apply' goal?

File: tf-build-tools/tf-cmd-api/src/main/java/com/deliveredtechnologies/terraform/api/TerraformApply.java [hunk 2]
```diff
@@ -87,23 +89,27 @@ public String execute(Properties properties) throws TerraformException {
 
     for (TerraformApplyParam param : TerraformApplyParam.values()) {
       if (properties.containsKey(param.property)) {
-        if (param == TerraformApplyParam.tfVarFiles) {
-          for (String file : (properties.getProperty(param.property)).split(",")) {
-            options.append(String.format("-%1$s=%2$s ", param, file.trim()));
+        if (!properties.containsKey(TerraformApplyParam.plan.property)) {
```
Comment (Clayton7510) [inline tf-build-tools/tf-cmd-api/src/main/java/com/deliveredtechnologies/terraform/api/TerraformApply.java:92]: This is redundant. Like 116 accounts for this scenario.

File: tf-build-tools/tf-cmd-api/src/main/java/com/deliveredtechnologies/terraform/api/TerraformDestroy.java [hunk 3]
```diff
@@ -23,6 +23,7 @@
     tfVars("var"),
     tfVarFiles("var-file"),
     noColor("no-color"),
+    refreshState("refresh"),
```
Comment (Clayton7510) [inline tf-build-tools/tf-cmd-api/src/main/java/com/deliveredtechnologies/terraform/api/TerraformDestroy.java:26]: Refresh will not be added to the terraform command. Please add a test for this to TerraformDestroyTest to validate when it is added.

File: README.md (hunk 1 above)
Comment (maganti07) [reply README.md]: Removed plan and refreshState in tf:destroy which is not required for terraform destroy.

File: tf-build-tools/tf-cmd-api/src/main/java/com/deliveredtechnologies/terraform/api/TerraformDestroy.java (hunk 3 above)
Comment (maganti07) [reply tf-build-tools/tf-cmd-api/src/main/java/com/deliveredtechnologies/terraform/api/TerraformDestroy.java]: Removed refreshState

**Pull request merged**